developing/nix/README.md

```diff
@@ -77,48 +77,211 @@ docker and a native development environment.
+PATH.
+
+### Extra Setup (Only Fish Shell Users)
+
+If you're using the fish shell, check out
+[nix-env.fish](https://github.com/lilyball/nix-env.fish).
+
+Add this to your `~/.config/fish/config.sh`
 
     set -g fish_user_paths "/Users/[your_os_username]/.nix-profile/bin" $fish_user_paths
 
-Now you want to create a new profile and install packages in that
-profile. If you have a directory `nix` with a `default.nix` in it, you
-can do
+### Working With Packages
+
+This is more of a quick overview, but more details can be found in the
+[nix basic package management docs](https://nixos.org/manual/nix/stable/#ch-basic-package-mgmt).
+
+#### Installing New Packages
+
+```shell
+nix-env -i <package>
+```
+
+E.g.
+
+```shell
+nix-env -i direnv
+```
+
+#### Seeing Installed Packages
+
+```shell
+nix-env -q
+```
+
+#### Looking for Packages
+
+```shell
+nix-env -qa <package name>
+```
 
-    nix-env -p /nix/var/nix/profiles/myproject -f ./nix -i
+E.g.
+
+```shell
+nix-env -qa python3
+```
 
-Create an `.envrc` file in a directory with some environment variables your project needs:
+#### Uninstalling a Package
 
+```shell
+nix-env -e <package>
 ```
+
+E.g.
+
+```shell
+nix-env -e chamber
+```
+
+### Profiles
+
+Profiles can be used at a global level, on a per-project basis, or
+on a per-repo basis.
+
+E.g. for MilMove, you might want a profile that contains things like
+`aws-vault`, `chamber`, etc. which spans more than just the `mymove`
+repo. On the other hand, you may not want to use a global profile
+(like the default one mentioned in the next section) because you may
+need different versions of `aws-vault` for different projects.
+
+#### Default
+
+When it's installed, `nix` creates a profile which will store all the
+packages you install by default, unless you tell it to use a different
+profile when you're installing a package, or you switch the active
+profile to be another one.
+
+Default profile is in: `/nix/var/nix/profiles/per-user/<username>/profile/`
+E.g. `/nix/var/nix/profiles/per-user/felipe/profile/`
+
+#### Creating a Profile
+
+You can create a new profile wherever you want, though convention seems
+to be to do so in
+`/nix/var/nix/profiles/`
+or
+`/nix/var/nix/profiles/per-user/<username>/`
+
+To create a new profile, you can run the command below. Note that this
+won't actually set this new profile as the active one, that's in the
+next section.
+
+```shell
+nix-env -p /nix/var/nix/profiles/<profile name> -i nix nss-cacert
+```
+
+E.g.
+
+```shell
+nix-env -p /nix/var/nix/profiles/milmove -i nix nss-cacert
+```
+
+The `-p` is telling it the path to the profile you want to use for
+this `nix` command. This can be used to work with profiles that aren't
+the active profile. It can also be used with profiles that don't yet
+exist (like here) or with exising profiles.
+
+You may remember the `-i` is used for installing, so we're installing
+`nix` and `nss-cacert` into our new profile.
```

Comment:
I recommended this at one time, but now I'm not so sure. You can have multiple profiles in your `PATH` and I think that's the better approach: to have the project profile first in `PATH` and the default `~/.nix-profile/bin` later in the `PATH`. That way you can have `nix` and friends in the default profile only.

Also note that you can set `NIX_PROFILE` to point to the current profile path so you don't need to pass `-p`.

Reply:
I'm seeing this a bit differently than you I think. Like I'm seeing this as people being able to have 3 different kinds of profiles:

1. Global profile, which is the default set up by `nix` when it's installed at `/nix/var/nix/profiles/per-user/<username>/profile/`
2. Global project profile, for common things needed across repos for a project. 
    1. E.g. for Milmove, we have multiple repos that each have their own dependencies, but several have some similar base requirements like `aws-vault`, `chamber`, etc.
    2. These can live at 
        `/nix/var/nix/profiles/`
        or
        `/nix/var/nix/profiles/per-user/<username>/`
3. Repo profiles. These would be defined in the repo with the `nix/default.nix` and use env vars set up in `.envrc`

My main reason for distinguishing between 1 and 2 above is that some people may work on other things locally that would need the same tools I define in 2, but different versions. So that would be a use case for having a "global project profile" so that if you are working on a project that uses different versions of tools for their repos, you can just switch global profiles and be set. Then the repo profiles kick in once you get into the repo directory. Maybe I'm over-complicating this though since I'm not sure there are many people that would need to work on different projects. 

Though I do like the idea of not having to install `nix` and `nss-cacert` into all profiles so if we keep the 3 profiles separated like I was thinking, I could update it to include guidance on having your regular project global inserted before the default profile in the `PATH`.

Reply:
IMHO, trying to solve the global project profile (use case 2) is a need that we haven't really proven yet. If we had every repo have its own nix config, yes we would have to update multiple repos when a new version of e.g. `aws-vault` came out, but I don't think we have enough experience to know if the additional complexity is worth it.

Maybe with more time we will know more.

Thoughts?

Reply:
Yeah, i'm good with keeping it simpler and then re-visit if we come across more concrete needs to change things.

Reply:
How about we move this content to an "advanced users" section or something with a preface about breaking things and invalidating your warranty? 😁

Reply:
That sounds like a good plan. I'll move it instead of deleting it and simplify this part.